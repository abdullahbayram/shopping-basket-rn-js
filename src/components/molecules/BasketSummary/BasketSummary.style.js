import { StyleSheet } from 'react-native';
import { spacing } from '@constants/theme';

export default StyleSheet.create({
  totalContainer: {
    marginVertical: spacing.md,
  },
});
