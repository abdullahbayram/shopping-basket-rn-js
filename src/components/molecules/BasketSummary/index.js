import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { strings } from '@constants';
import styles from './BasketSummary.style';

const BasketSummary = ({ totalItemCount, totalPrice }) => {
  return (
    <View>
      {totalItemCount && (
        <Text variant="titleSmall">
          {strings.payment.basketItemCount} {totalItemCount}
        </Text>
      )}
      <View style={styles.totalContainer}>
        <Text variant="titleSmall">
          {strings.payment.total} ${Number.isNaN(totalPrice) ? '0.00' : totalPrice.toFixed(2)}
        </Text>
      </View>
    </View>
  );
};

BasketSummary.propTypes = {
  totalItemCount: PropTypes.number,
  totalPrice: PropTypes.number.isRequired,
};

export default BasketSummary;
