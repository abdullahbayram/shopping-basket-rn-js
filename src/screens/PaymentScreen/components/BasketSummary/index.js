import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { strings } from '@constants';
import styles from './BasketSummary.style';

const BasketSummary = ({ totalCount, total }) => {
  return (
    <View>
      <Text variant="titleSmall">
        {strings.payment.basketItemCount} {totalCount}
      </Text>
      <View style={styles.totalContainer}>
        <Text variant="titleSmall">
          {strings.payment.total} ${Number.isNaN(total) ? '0.00' : total.toFixed(2)}
        </Text>
      </View>
    </View>
  );
};

BasketSummary.propTypes = {
  totalCount: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
};

export default BasketSummary;
