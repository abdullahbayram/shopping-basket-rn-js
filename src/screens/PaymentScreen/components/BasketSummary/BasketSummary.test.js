import React from 'react';
import { render, screen } from '@testing-library/react-native';
import BasketSummary from '.';

describe('<BasketSummary />', () => {
  const mockTotalCount = 14;
  const mockTotal = 2648.01;

  it('renders correctly and matches the snapshot', () => {
    const { toJSON } = render(<BasketSummary totalCount={mockTotalCount} total={mockTotal} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('displays the correct totalCount and total', () => {
    render(<BasketSummary totalCount={mockTotalCount} total={mockTotal} />);

    expect(screen.getByText(`Items in the basket: ${mockTotalCount}`)).toBeTruthy();
    expect(screen.getByText(`Total: $${mockTotal.toFixed(2)}`)).toBeTruthy();
  });

  it('renders default value when total is NaN', () => {
    render(<BasketSummary totalCount={mockTotalCount} total={NaN} />);

    expect(screen.getByText('Total: $0.00')).toBeTruthy();
  });
});
