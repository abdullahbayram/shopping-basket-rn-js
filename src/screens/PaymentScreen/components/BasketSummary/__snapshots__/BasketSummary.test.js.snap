// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketSummary /> renders correctly and matches the snapshot 1`] = `
<View>
  <Text
    variant="titleSmall"
  >
    Items in the basket: 
     
    14
  </Text>
  <View
    style={
      {
        "marginVertical": 16,
      }
    }
  >
    <Text
      variant="titleSmall"
    >
      Total:
       $
      2648.01
    </Text>
  </View>
</View>
`;
